File: typescript/11. Container With Most Water.ts
function maxArea(height: number[]): number {
  let l = 0;
  let r = height.length - 1;
  let res = 0;
  while (l < r) {
    const minHeight = Math.min(height[l], height[r]);
    const area = minHeight * (r - l);

    res = Math.max(res, area);

    if (height[l] < height[r]) {
      l++;
    } else {
      r--;
    }
  }

  return res;
}

File: golang/daily/11. Container With Most Water.go
package daily

func maxArea(height []int) int {
	l := 0
	r := len(height) - 1

	var max int
	for l < r {
		amount := (r - l) * min(height[r], height[l])
		if amount > max {
			max = amount
		}

		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}
	return max
}

func min(a, b int) int {
	if a < b {
		return a
	} else {
		return b
	}
}
